core/node/builder.go

```diff
@@ -40,6 +41,9 @@ type BuildCfg struct {
 	Routing libp2p.RoutingOption
 	Host    libp2p.HostOption
 	Repo    repo.Repo
+
+	// Bitswap
+	PeerBlockRequestFilter bitswap.PeerBlockRequestFilter
```

Comment:
I'm not sure which we should prefer:
1. Pass the `PeerBlockRequestFilter` function, shown here,
2. Pass a generic `bitswap.Option` function.

(2) is more "obviously future-proof", but that'd mean we'd have two different ways to configure most values in the bitswap instance  (json from cfg and code from buildcfg).

Access control is probably a cross-cutting concern that will span multiple modules and protocols, not only bitswap. So by decoupling the access control config from the bitswap option, (1) might make it easier to build upon this feature.